fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/base/ResourceAmount.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.storage.base;
+
+import java.util.Objects;
+
+/**
+ * An immutable object storing both a resource and an amount, provided for convenience.
+ * @param <T> The type of the stored resource.
+ */
+public class ResourceAmount<T> {
```

Comment:
if this targets 1.17, can just use a record

Reply:
Good point, it will target 1.17 I think.

Reply:
If it would target 1.17, it would have to be backported, so I have decided to keep this PR on 1.16 for the time being (and thus java 8). Marking as resolved.